plugins/by-name/trouble/default.nix

```
@@ -115,200 +115,274 @@ helpers.neovim-plugin.mkNeovimPlugin {
+          symbols = {
+            desc = "document symbols";
+            mode = "lsp_document_symbols";
+            focus = false;
+            win = {
+              position = "right";
+            };
+            filter = {
+              "not" = {
+                ft = "lua";
+                kind = "Package";
+              };
+              any = {
+                ft = [
+                  "help"
+                  "markdown"
+                ];
+                kind = [
+                  "Class"
+                  "Constructor"
+                  "Enum"
+                  "Field"
+                  "Function"
+                  "Interface"
+                  "Method"
+                  "Module"
+                  "Namespace"
+                  "Package"
+                  "Property"
+                  "Struct"
+                  "Trait"
+                ];
+              };
+            };
+          };
+        }
+        ''
+          Customize the various builtin modes or define your own.
+        '';
 
-    use_diagnostic_signs = helpers.defaultNullOpts.mkBool false ''
-      Enabling this will use the signs defined in your lsp client
-    '';
+    icons = defaultNullOpts.mkAttrsOf lib.types.anything {
+      indent = {
+        top = "│ ";
+        middle = "├╴";
+        last = "└╴";
+        fold_open = " ";
+        fold_closed = " ";
+        ws = "  ";
+      };
+      folder_closed = " ";
+      folder_open = " ";
+      kinds = {
+        Array = " ";
+        Boolean = "󰨙 ";
+        Class = " ";
+        Constant = "󰏿 ";
+        Constructor = " ";
+        Enum = " ";
+        EnumMember = " ";
+        Event = " ";
+        Field = " ";
+        File = " ";
+        Function = "󰊕 ";
+        Interface = " ";
+        Key = " ";
+        Method = "󰊕 ";
+        Module = " ";
+        Namespace = "󰦮 ";
+        Null = " ";
+        Number = "󰎠 ";
+        Object = " ";
+        Operator = " ";
+        Package = " ";
+        Property = " ";
+        String = " ";
+        Struct = "󰆼 ";
+        TypeParameter = " ";
+        Variable = "󰀫 ";
+      };
+    } "Define custom icons.";
   };
 
   extraConfig = cfg: {
+    # TODO: introduced 2024-10-12: remove after 24.11
+    warnings =
+      let
+        definedOpts = lib.filter (opt: lib.hasAttrByPath (lib.toList opt) cfg.settings) [
+          "action_keys"
+          "position"
+          "height"
+          "width"
+          "mode"
+          "fold_open"
+          "fold_close"
+          "group"
+          "padding"
+          "cycle_results"
+          "indent_lines"
+          "win_config"
+          "include_declaration"
+          "signs"
+          "use_diagnostic_signs"
+        ];
+      in
+      lib.optional (definedOpts != [ ]) ''
+        Nixvim(plugins.trouble): The following v2 settings options are no longer supported in v3:
+        ${lib.concatMapStringsSep "\n" (opt: "  - ${lib.showOption (lib.toList opt)}") definedOpts}
```

Which allows simplifying this too:
```suggestion
        ${lib.concatMapStringsSep "\n" (opt: "  - ${opt}") definedOpts}
```